src/EditorFeatures/Core/Intents/DeleteParameterIntentProvider.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ChangeSignature;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Features.Intents;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Rename;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.EditorFeatures.Intents;
+
+[IntentProvider(WellKnownIntents.DeleteParameter, LanguageNames.CSharp), Shared]
+internal class DeleteParameterIntentProvider : IIntentProvider
+{
+    private readonly IGlobalOptionService _globalOptionService;
+
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    public DeleteParameterIntentProvider(IGlobalOptionService globalOptionService)
+    {
+        _globalOptionService = globalOptionService;
+    }
+
+    public async Task<ImmutableArray<IntentProcessorResult>> ComputeIntentAsync(
+        Document priorDocument,
+        TextSpan priorSelection,
+        Document currentDocument,
+        IntentDataProvider intentDataProvider,
+        CancellationToken cancellationToken)
+    {
+        var changeSignatureService = priorDocument.GetRequiredLanguageService<AbstractChangeSignatureService>();
+        var contextResult = await changeSignatureService.GetChangeSignatureContextAsync(
+            priorDocument, priorSelection.Start, restrictToDeclarations: false, _globalOptionService.CreateProvider(), cancellationToken).ConfigureAwait(false);
```

Comment:
what does the restructToDeclarations flag do?

Reply:
That means it would only provide the context if invoked on a declaration.  We want it on both